src/python/dxpy/scripts/dx.py

```diff
@@ -2869,6 +2880,80 @@ def check_for_special_access(access_spec):
     else:
         run_batch_all_steps(args, executable, dest_proj, dest_path, input_json, run_kwargs)
 
+def get_merged_cluster_spec(app_reqs, instance_count):
+    '''
+    Returns a dictionary that can be passed as a "systemRequirements" input to app-xxx/run, e.g.
+    {'fn': {'clusterSpec': {initialInstanceCount: 3, version: "2.4.0", ...}}}
+
+    Since full clusterSpec must be passed to the API server, we need to retrieve the cluster
+    spec defined in app doc's systemRequirements and overwrite the field initialInstanceCount
+    with the value the user passed to dx run for each entrypoint.
+
+    Instance count is currently the only clusterSpec's field the user is allowed to change
+    at runtime.
+
+    A few scenarios when requesting instance count for different entrypoints with dx run 
+    and the resulting merged systemRequirements (merged_cluster_spec):
+
+    Requested: {"*": 5}
+    App doc: {"main": "clusterSpec": {"initialInstanceCount": 7, bootstrapScript: "x.sh"},
```

Comment:
not sure that it matters here, but type, version, & ports are also fields that would need to be copied over. I assume those are handled correctly as well.

Reply:
Yes, every other field is copied (in this deep copy [here](https://github.com/dnanexus/dx-toolkit/blob/53089e78f4f87fa951f3a5663ef74ad76fc53b88/src/python/dxpy/scripts/dx.py#L2943)), I'll add a note to the docstring